Sources/NIOPosix/SelectableEventLoop.swift

```diff
@@ -30,6 +30,27 @@ internal func withAutoReleasePool<T>(_ execute: () throws -> T) rethrows -> T {
 #endif
 }
 
+/// Information about an EventLoop tick
+public struct EventLoopTickInfo {
+    /// The number of tasks which were executed in this tick
+    public var numberOfTasks: Int
+    /// The time at which the tick began
+    public var startTime: NIODeadline
+
+    internal init(numberOfTasks: Int, startTime: NIODeadline) {
+        self.numberOfTasks = numberOfTasks
+        self.startTime = startTime
+    }
+}
+
+/// Implement this delegate to receive information about the EventLoop, such as each tick
+public protocol EventLoopDelegate {
```

Comment:
- needs a `NIO` prefix to satisfy the public API guarantees NIO has
- I'd call this `NIOEventLoopMetricsDelegate` or something